music/src/ddsp/pitch_utils.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * @license
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import * as tf from '@tensorflow/tfjs';
+import {
+  CONF_THRESHOLD,
+  PT_SLOPE,
+  PT_OFFSET,
+  PITCH_CONF_JITTER,
+} from './constants';
+import { AudioData } from './interfaces';
+import { Tensor } from '@tensorflow/tfjs';
+
+function upsample_linear(
+  spicePitchesOutput: number[],
```

Comment:
you could name these a bit more generally, because it's actually just a linear interpolation function right?